### PR TITLE
Fix release deps & docs

### PR DESCRIPTION
**Pull request by josvazg**
### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?

### CODE REVIEW

File: scripts/install-docker-sbom-plugin.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+download_url_base=https://github.com/docker/sbom-cli-plugin/releases/download
+url="${download_url_base}/v${version}/sbom-cli-plugin_${version}_${os}_${arch}.tar.gz"
+
+curl -L "${url}" -o "${target}"
+pushd "${TMPDIR}"
+tar zxvf "${target}" docker-sbom
+chmod +x docker-sbom
+popd
+sudo cp "${TMPDIR}/docker-sbom" "${docker_dir}"
```
Comment (s-urbaniak) [inline scripts/install-docker-sbom-plugin.sh:20]: Don't execute `sudo` commands in scripts.

**Review by s-urbaniak — CHANGES_REQUESTED** [review]
one concern around `sudo` in the script. I'd rather recommend leave it in documentation, maybe point to the script but definitely no automation in Makefile.

File: Makefile [hunk 2]
```diff
@@ -482,7 +484,15 @@ govulncheck:
 vulncheck: ## Run govulncheck to find vulnerabilities in code
 	@./scripts/vulncheck.sh ./vuln-ignore
 
+envsubst:
+	@which envsubst || go install github.com/drone/envsubst/cmd/envsubst@latest
+
+docker-sbom:
+	@docker sbom --help > /dev/null || \
+	DOCKER_SBOM_PLUGIN_VERSION=$(DOCKER_SBOM_PLUGIN_VERSION) \
+	ARCH=$(TARGET_ARCH) OS=$(TARGET_OS) ./scripts/install-docker-sbom-plugin.sh
```
Comment (s-urbaniak) [inline Makefile:493]: Let's not automate this. We can spit out a message "please install the docker sbom plugin" and point to the script.

File: scripts/install-docker-sbom-plugin.sh (hunk 1 above)
Comment (josvazg) [reply scripts/install-docker-sbom-plugin.sh:20]: I left this but instead removed the automation to call it, so it must be used manually.

File: Makefile (hunk 2 above)
Comment (josvazg) [reply Makefile]: I have replaced with a help message pointing to the docs, and the docs explain how to install and link to the script to be used manually if one prefers that.

File: scripts/install-docker-sbom-plugin.sh (hunk 1 above)
Comment (s-urbaniak) [reply scripts/install-docker-sbom-plugin.sh:20]: suggestion: let's prefix this script with sudo, i.e. `scripts/sudo-install-docker-sbom-plugin.sh` to make it very clear that this is entering dangerous territory.

File: docs/dev/release.md [hunk 3]
```diff
@@ -10,6 +10,13 @@ This is not required for [Certified Operators](https://github.com/redhat-openshi
 
 Finally, make sure you have a "RedHat Connect" account and are a [team member with org administrator role in the team list](https://connect.redhat.com/account/team-members).
 
+### Tools
+
+Most tools are automatically installed for you. Most of them are Go binaries and use `go install`. There are a few that might cause issues and you might want to pre-install manually:
+
+- [envsubst](https://www.gnu.org/software/gettext/manual/html_node/envsubst-Invocation.html) from the `gettext` package is used for expanding env vars on files or strings. Here is needed to autogenerate the SDLC checklist document from a template. You can install it manually with `brew` or `apt`/`dnf`, otherwise the makefile will automatically try to install a [envsubst drop-in replacement in Go](https://github.com/drone/envsubst?tab=readme-ov-file).
```
Comment (s-urbaniak) [inline docs/dev/release.md:17]:
```suggestion
- [envsubst](https://www.gnu.org/software/gettext/manual/html_node/envsubst-Invocation.html) for autogenerating the SDLC checklist. Install manually, else makefile automation will install it using `go install`.
```

File: docs/dev/release.md [hunk 4]
```diff
@@ -10,6 +10,13 @@ This is not required for [Certified Operators](https://github.com/redhat-openshi
...
+- [docker sbom](https://github.com/docker/sbom-cli-plugin/) is a plugin for docker, it might not work for you out of the box. It can be activated as experimental on Docker for Mac. You can pre-install it manually by [following the official instructions](https://github.com/docker/sbom-cli-plugin/?tab=readme-ov-file#getting-started) or could run something like [this script](../../scripts/install-docker-sbom-plugin.sh).
```
Comment (s-urbaniak) [inline docs/dev/release.md:18]:
```suggestion
- [Docker SBOM plugin](https://github.com/docker/sbom-cli-plugin/) for generating SBOM files. It is available as an experimental feature on Docker for Mac or can be installed manually by [following the official instructions](https://github.com/docker/sbom-cli-plugin/?tab=readme-ov-file#getting-started). A [helper install script](../../scripts/install-docker-sbom-plugin.sh). Note: the helper scripts executes command using `sudo` privileges.
```

**Review by s-urbaniak — CHANGES_REQUESTED** [review]
just a few nits else LGTM

File: docs/dev/release.md (hunk 4 above)
Comment (josvazg) [reply docs/dev/release.md:18]:
```suggestion
- [Docker SBOM plugin](https://github.com/docker/sbom-cli-plugin/) for generating SBOM files. It is available as an experimental feature on Docker for Mac or can be installed manually by [following the official instructions](https://github.com/docker/sbom-cli-plugin/?tab=readme-ov-file#getting-started). Another option is a [helper install script](../../scripts/install-docker-sbom-plugin.sh). Note: the helper scripts executes command using `sudo` privileges.
```

File: scripts/install-docker-sbom-plugin.sh (hunk 1 above)
Comment (josvazg) [reply scripts/install-docker-sbom-plugin.sh:20]: will do

**Review by s-urbaniak — APPROVED** [review]

**Pull request merged**